libportability-gfx/src/conv.rs

```diff
@@ -170,9 +170,17 @@ pub fn map_subresource_range(subresource: VkImageSubresourceRange) -> image::Sub
     image::SubresourceRange {
         aspects: map_aspect(subresource.aspectMask),
         levels: subresource.baseMipLevel as _
-            ..(subresource.baseMipLevel + subresource.levelCount) as _,
+            .. if subresource.levelCount as i32 == VK_REMAINING_MIP_LEVELS {
+                !0
```

Comment:
Where do we handle the `!0` cases on the HAL side currently? For example, in the Metal backend we seem to iterate directly through this range without special handling for this case

Reply:
Metal does actually handle `!0` in `clear_image` (at least). It's not pretty, and I'm open to ideas on how to expose this better. One option would be to just always require the exact level/layer counts, and our portability layer can remember them.